### ISSUE TITLE
UnicodeDownload data is garbled by older version of links

### ISSUE THREAD

**Issue by flwyd** [chrisbra/unicode.vim#52]
## TL;DR

When netrw prefers `links` to `wget` or `curl`, `:UnicodeDownload` saves some binary data, not the plain text UnicodeData.txt file.

I suspect that on my system `links -source` is saving the compressed or encrypted https data rather than the underlying content.  However, `file` isn't able to identify what format `:UnicodeDownload` saved, and it's 412kB vs. 1.8MB for the uncompressed UnicodeData.txt file and the 276kB gzip-compressed result when I run `links $URL -source > /tmp/unicode.txt`

The plugin doesn't report an error when this happens, leading to confusing vim exceptions latter on when trying to use unicode plugin features.
I suggest that after `:UnicodeDownload` completes, it should call `UnicodeDict()` and, if there's an error, inform the user that the automatic download was unsuccessful.  This may be more helpful to users than the "if this doesn't work" message prior to attempting the download, since the download might happen long before using the plugin (e.g. if installing via `Plug 'chrisbra/unicode.vim', { 'do': ':UnicodeDownload' }`).

## Details

Vim and links versions on UbuntuLinux 18.04.5 LTS:
```
VIM - Vi IMproved 8.0 (2016 Sep 12, compiled Oct 13 2020 15:49:09)
Included patches: 1-1453
Package: links
Version: 2.14-5build1
Built-Using: publicsuffix (= 20180125.0922-1)
```

I get the following error when using most unicode features after running `:UnicodeDownload`.
```
Error detected while processing function unicode#CompleteUnicode[5]..<SNR>234_UnicodeDict:
line   16:
E684: list index out of range: 1
```
Line 16 of `UnicodeDict` is splitting a line on `;` but the data file that got downloaded is a binary mess:
```
00000000  1f c2 8b 08 00 00 00 00  00 00 03 c2 a4 c3 bd 59  |...............Y|
00000010  c2 93 c3 a4 3a 76 25 0a  c2 bf c3 bb c2 af c2 88  |....:v%.........|
00000020  c3 87 c3 bb 3d c2 a4 5d  c3 a7 4c 5a c3 88 c3 9a  |....=..]..LZ....|
00000030  0c 24 41 77 66 c3 b8 c2  94 c3 ae c3 b4 c2 8c 13  |.$Awf...........|
…
```

Fetching https://www.unicode.org/Public/UNIDATA/UnicodeData.txt from that system using `curl` works fine, so I think this is a result of my system's `netrw.vim` preferring `links` over `curl` or `wget`, though this also produces a different output than `links -source`.

```
:UnicodeDownload
UnicodePlugin: Updating /home/tstone/.vim/plugged/unicode.vim/autoload/unicode/UnicodeData.txt
UnicodePlugin: Let's see, if we can download it.
UnicodePlugin: If this doesn't work, you should download 
UnicodePlugin: https://www.unicode.org/Public/UNIDATA/UnicodeData.txt and save it as /home/tstone/.vim/plugged/unicode.vim/autoload/unicode/UnicodeData.txt
:echo g:netrw_http_cmd
links
:echo g:netrw_http_xcmd
-source >
```

**Comment by bpj** [comment]
I think I found a typo. See comment on the commit.

-- 
Better --help|less than helpless

Den tors 19 nov. 2020 09:11Christian Brabandt <notifications@github.com>
skrev:

**Comment by chrisbra** [comment]
yes, I pushed too fast. I think it is fixed in recent master.
